build.gradle

```diff
@@ -639,7 +640,7 @@ dependencies {
     integrationTestImplementation 'com.unboundid:unboundid-ldapsdk:4.0.14'
 
     //Checkstyle
-    checkstyle 'com.puppycrawl.tools:checkstyle:10.12.1'
+    implementation "com.puppycrawl.tools:checkstyle:${checkstyle_version}"
```

Comment:
I see that we also have an entry in the gradle plugins section that adds the [checkstyle gradle plugin](https://github.com/opensearch-project/security/blob/main/build.gradle#L65).

Could this entry in the dependencies section be redundant?

Reply:
@cwperks it seems likely, I removed the line 643 and it worked.
should I remove it and rebase the PR or is there any other way I can test it?